relion/tests/test_projection_subtraction_no_relion.py

```diff
@@ -0,0 +1,165 @@
+# ***************************************************************************
+# * Authors:    Marta Martinez (mmmtnez@cnb.csic.es)
+# *             Roberto Marabini (roberto@cnb.csic.es)
+# *
+# *
+# * This program is free software; you can redistribute it and/or modify
+# * it under the terms of the GNU General Public License as published by
+# * the Free Software Foundation; either version 2 of the License, or
+# * (at your option) any later version.
+# *
+# * This program is distributed in the hope that it will be useful,
+# * but WITHOUT ANY WARRANTY; without even the implied warranty of
+# * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# * GNU General Public License for more details.
+# *
+# * You should have received a copy of the GNU General Public License
+# * along with this program; if not, write to the Free Software
+# * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+# * 02111-1307  USA
+# *
+# *  All comments concerning this program package may be sent to the
+# *  e-mail address 'scipion@cnb.csic.es'
+# ***************************************************************************/
+""" test subtraction projection protocol when relion
+ alignment is not available"""
+from pwem.protocols import ProtImportParticles
+from pyworkflow.tests import *
+from tempfile import NamedTemporaryFile
+
+from relion.protocols import ProtRelionSubtract
+
+
+class TestSubtractionProjection(BaseTest):
+
+    @classmethod
+    def setUpClass(cls):
+        setupTestProject(cls)
+        cls.xmippAvailable = True
+        cls.sampling = 1.38
+        from pwem import Domain
+        try:
+            cls.xmipp3 = \
+                Domain.importFromPlugin('xmipp3', doRaise=True)
```

Comment:
no new xmipp-invoking code is accepted in this plugin, so this whole test will not be accepted

Reply:
I will delete the test if you wish but I think it is  better than the one implemented for subtract_projection. I tried to use the same data set  that you use but the result was confusing and it was not clear if the protocol worked or did not.

At present, this test will not fail if xmipp is not installed, it will just be skipped,  and  it is easy to understand the results.
Personally I like this kind of tests that do not rely in external data sets. I guess we do not share the same point of view
and you prioritize tests that do not rely on other packages. So if you want I will delete it.

Reply:
will it be acceptable to incorporate this code as a workflow instead of as a test?

Reply:
I agree to follow @delarosatrevin decision, if he agrees for this test to be in the plugin. If yes, it should be in a separate file (as it is now), so I won't ever touch or look or support it. Also, if the plugin changes in the future I will not update this test code.

Reply:
Thank you both @rmarabini and @azazellochg.  I think we can keep the test if xmipp is optional and it is not troublesome.  If in the future it requires too much maintenance, we can re-consider if keeping it or not.